DexPriceAggregatorUniswapV3/contracts/DexPriceAggregatorUniswapV3.sol

```diff
@@ -0,0 +1,409 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "./interfaces/IUniswapV3Pool.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/math/SafeCast.sol";
+
+import "./libraries/PoolAddress.sol";
+import "./libraries/OracleLibrary.sol";
+import "./interfaces/IDexPriceAggregator.sol";
+
+/// @title DexPriceAggregatorUniswapV3
+/// @notice A DexPriceAggregator sourcing prices from Uniswap V3.
+///         Provides the minimum output between an asset's "spot" price and TWAP from the last n seconds.
+///         The "spot" price is always the last block's ending price.
+contract DexPriceAggregatorUniswapV3 is IDexPriceAggregator, Ownable {
+    address public immutable uniswapV3Factory;
+    address public immutable weth;
```

Comment:
A public state variable is equivalent to a function for the purposes of NatSpec.

Solidity NatSpec Documentation
https://docs.soliditylang.org/en/v0.8.17/natspec-format.html